roles/edpm_nova/scripts/derive_pci_passthrough_whitelist.py

```diff
@@ -0,0 +1,587 @@
+    :return user_config_copy: Any non-nic partinioned cfg will be returned in this list
+    :return derived_config: All nic partinioned cfg will be returned after derivation in this list
+    """
+
+    user_config_copy = []
+    derived_config = []
+
+    allocated_pci = get_allocated_pci_addresses(system_configs)
+    nic_partition_pfs = get_sriov_nic_partition_pfs(system_configs)
+
+    for user_config in user_configs:
+        if 'address' in user_config and 'devname' in user_config:
+            msg = f"Both devname and address can't be present in {_PASSTHROUGH_WHITELIST_KEY}"
+            raise InvalidConfigException(msg)
+
+        keys = ['address', 'product_id', 'devname']
+        if not any(k in user_config for k in keys):
+            # address or product_id or devname not present in user_config
+            pf = get_pf_name_from_phy_network(user_config['physical_network'])
+            user_config['address'] = get_pci_addr_from_ifname(pf)
+
+        if 'devname' in user_config:
+            if user_config['devname'] in nic_partition_pfs:
+                user_config['address'] = get_pci_addr_from_ifname(user_config['devname'])
+                del user_config['devname']
+            else:
+                user_config_copy.append(user_config)
+                continue
+        if ('address' in user_config
+                or 'product_id' in user_config):
+            passthrough_tmp, del_user_config = get_passthrough_config_for_all_pf(
+                user_config, system_configs, allocated_pci)
+
+            """ If del_user_config is set, do not add to derived_config or
+                user_config_copy
+            """
+            if not del_user_config:
+                if passthrough_tmp:
+                    derived_config.extend(passthrough_tmp)
+                else:
+                    user_config_copy.append(user_config)
+        else:
+            user_config_copy.append(user_config)
+    return (user_config_copy, derived_config)
+
+
+def get_nova_userconfigs(nova_conf_file):
+    configs = []
+    src_config = ""
+    # Currently supporting both formats
+    valid_spec = ['passthrough_whitelist', 'device_spec']
+
+    # Read user config from edpm_nova_config conf file
+    try:
+        with open(nova_conf_file) as nova_config:
+            for line in nova_config:
+                pf = line.strip()
+                if pf.startswith(tuple(valid_spec)):
+                    # make a copy for replace after derive
+                    src_config += line
+                    obj = pf.split('=')
+                    if isinstance(obj, str):
+                        configs.append(json.loads(obj[1]))
+                    else:
+                        configs.append(obj[1])
+    except IOError:
+        configs = []
+    return (configs, src_config)
+
+
+if __name__ == "__main__":
+    pci_passthro = {}
+    sriov_nova_conf_file = '/var/lib/openstack/config/nova/03-sriov-nova.conf'
```

Comment:
@vcandapp Can we change the file name used similar like VA architecture.

Reply:
This will be take as user input